tests/unit/QueryComponentFactoryTest.php

```diff
@@ -72,19 +87,104 @@ public function testEscapeTerms()
+
+        $criteriaValue = Criteria::where('TestField')->is('TestValue');
+        $criteria = $reflectionMethod->invoke(new QueryComponentFactory(), 'TestField', $criteriaValue);
+        $this->assertInstanceOf(Criteria::class, $criteria);
+        $this->assertSame($criteriaValue, $criteria);
+    }
 
-    protected function setUp()
+    public function testFilterAndExcludeStrings()
     {
-        parent::setUp();
-        Injector::inst()->get(Page::class)->requireDefaultRecords();
-        foreach (self::$extra_dataobjects as $className) {
-            Config::modify()->merge($className, 'extensions', [DataObjectExtension::class]);
-        }
-        $this->factory = new QueryComponentFactory();
-        $this->factory->setIndex(Injector::inst()->get(CircleCITestIndex::class));
+        $mockFilterQuery = $this->createMock(FilterQuery::class);
+        $mockFilterQuery->expects($this->once())
+            ->method('setQuery')
+            ->with($this->equalTo('TestFilterField:TestFilterValue'));
+        $mockExcludeQuery = $this->createMock(FilterQuery::class);
+        $mockExcludeQuery->expects($this->once())
+            ->method('setQuery')
+            ->with($this->equalTo('-TestExcludeField:TestExcludeValue'));
+
+        $mockClientQuery = new class extends Query {
+            public $mockFilterQuery;
+            public $mockExcludeQuery;
+            public function createFilterQuery($options = null): FilterQuery
+            {
+                if ($options === 'filter-TestFilterField') {
+                    return $this->mockFilterQuery;
+                } elseif ($options === 'exclude-TestExcludeField') {
+                    return $this->mockExcludeQuery;
+                }
+
+                return parent::createFilterQuery($options);
+            }
+        };
+        $mockClientQuery->mockFilterQuery = $mockFilterQuery;
+        $mockClientQuery->mockExcludeQuery = $mockExcludeQuery;
+
+        $baseQuery = new BaseQuery();
+        $baseQuery->addFilter('TestFilterField', 'TestFilterValue');
+        $baseQuery->addExclude('TestExcludeField', 'TestExcludeValue');
+
+        $factory = new QueryComponentFactory();
+        $factory->setIndex(Injector::inst()->get(CircleCITestIndex::class));
+        $factory->setQuery($baseQuery);
+        $factory->setClientQuery($mockClientQuery);
+        $factory->buildQuery();
+    }
+
+    public function testFilterAndExcludeCriteria()
+    {
+        $mockFilterQuery = $this->createMock(FilterQuery::class);
+        $mockFilterQuery->expects($this->once())
+            ->method('setQuery')
+            ->with($this->equalTo('TestFilterField:TestFilterValue'));
+        $mockExcludeQuery = $this->createMock(FilterQuery::class);
+        $mockExcludeQuery->expects($this->once())
+            ->method('setQuery')
+            ->with($this->equalTo('-TestExcludeField:TestExcludeValue'));
+
+        $mockClientQuery = new class extends Query {
```

Comment:
Using an anonymous class here was actually much easier than the mock objects I started out with... because `createFilterQuery()` is called an unknown number of times before it reaches the part I needed to mock, using a PHPUnit mock is very brittle (any code changes which alter how many times `createFilterQuery()` is called would break the tests)

Reply:
Merged, thanks for your great contribution. @Firesphere went ahead and pull the trigger :)